docker_fetch_base_cloudimg.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+# Helper to fetch the given ubuntu cloud image, and verify its signatures against the key file in
+# the repository.  Requires mktemp, gpg2/gpgv2, wget
+#
+# The ultimate effect of this script is roughly:
+#   wget https://partner-images.canonical.com/core/unsupported/precise/current/ubuntu-precise-core-cloudimg-${ARCH}-root.tar.gz
+#   wget https://partner-images.canonical.com/core/unsupported/precise/current/SHA256SUMS
+#   wget https://partner-images.canonical.com/core/unsupported/precise/current/SHA256SUMS.gpg
+#
+# ... Followed by the necessary gpg and sha256sum commands to ensure the image checksum & signature
+# matches ./ubuntu-cloud-key.txt (see verify() below)
+#
```

Comment:
Does this need to be reworked to match the new imported file ubuntu-archive-keyring.gpg instead? CC @smcv

Reply:
If that keyring has the cloud signing key in it, we can skip the lines to make a keyring for gpgv and just reference it.  Though if we're adding the cloud signing key to the same bundle, we may be teaching apt more keys than it should expect to see on package signatures.

Reply:
No, the cloud image signing key and the archive signing key are probably different - at least, Debian uses different keys to sign CD images and the apt archive, I'd expect Ubuntu to do similar.

We want to assert that apt archives (Release.gpg, really) are signed with the (old) archive signing key, and that cloud images are signed with the cloud image signing key.